tern/analyze/common.py

```diff
@@ -249,6 +317,11 @@ def add_base_packages(image_layer, binary, shell):
             shell = constants.shell
         # get all the packages in the base layer
         pkg_dict, invoke_msg, warnings = collate_list_metadata(shell, listing)
+
+        if listing.get("pkg_format") == "deb":
+            pkg_dict["licenses"] = get_deb_package_licenses(
```

Comment:
Here's where our logic fails: we're assuming that each package has 1 declared license, but for Debian, one package can have many licenses. We would like to associate the licenses to the package it came from. How do we do that if we can only store 1 license string in the `Package` class?

Reply:
I would suggest to create a new property in `Package` called "licenses" to store multiple licenses if they can be found.

Reply:
> Here's where our logic fails: we're assuming that each package has 1 declared license, but for Debian, one package can have many licenses. We would like to associate the licenses to the package it came from. How do we do that if we can only store 1 license string in the `Package` class?

I can get your point here that is why the previous commits in this PR collect all the licenses of a package and assign `pkg_license` property a string of all licenses collected. So say if a package `dpkg` got two licenses `['GPL-V2, LGPL V3+']` then the package's  `pkg_license` property would be having a string value `GPL-V2, LGPL V3+` which means a package would be associated with its respective licenses.

Reply:
> I would suggest to create a new property in `Package` called "licenses" to store multiple licenses if they can be found.

I have incorporated this change in the latest commit. Now I am having `pkg_licenses` property in the `Package` class and storing package licenses inside it. This change is also required to update the default report generator to print `pkg_licenses` if we have any so made that change also in the latest commit.